### PR TITLE
#25 technical debt

### PR DESCRIPTION
**Pull request by Seetaramayya**

### CODE REVIEW

File: src/main/scala/performanceanalysis/administrator/AdministratorActor.scala [hunk 1]
```diff
@@ -15,21 +15,27 @@ object AdministratorActor {
...
     case GetDetails(componentId) =>
-      handleGetDetails(logParserActors, componentId, sender)
+      handleGetDetails(logParserActors, componentId)
     case GetRegisteredComponents =>
-      ???
+      handleRegisteredComponents(logParserActors)
   }
 
-  private def handleRegisterComponent(logParserActors: Map[String, ActorRef], componentId: String, sender: ActorRef) = {
+  private def handleRegisteredComponents(logParserActors: Map[String, ActorRef]) = {
```
Comment (jordi133) [inline src/main/scala/performanceanalysis/administrator/AdministratorActor.scala]: Wouldn't the name `handleGetRegisteredComponents` be more descriptive than `handleRegisteredComponents`? And you forgot to add unit tests for this.

Comment (Seetaramayya) [inline src/main/scala/performanceanalysis/administrator/AdministratorActor.scala]: I agree I'll change it also I'll add unit test

File: src/test/scala/performanceanalysis/administrator/AdministratorActorSpec.scala [hunk 2]
```diff
@@ -54,6 +57,25 @@ class AdministratorActorSpec(_system: ActorSystem) extends ActorSpecBase(_system
...
+      4 to 10 foreach { i =>
+        val componentName: String = s"newComponent$i"
+        testProbe.send(adminActor, RegisterComponent(componentName))
+        testProbe.expectMsg(LogParserCreated(`componentName`))
+      }
+
+      testProbe.send(adminActor, GetRegisteredComponents)
+
+      val expectedComponents:Set[String] = ((4 to 10) map (i => s"component$i")).toSet
+      testProbe.expectMsgPF() { case RegisteredComponents(actualComponents) => (expectedComponents -- actualComponents).isEmpty }
```
Comment (TimSoethout) [inline src/test/scala/performanceanalysis/administrator/AdministratorActorSpec.scala:71]: I'm not sure if this line actually asserts something.

`expectedComponents should contain allOf(actualComponents)` <- using `Matchers`

Comment (TimSoethout) [inline src/test/scala/performanceanalysis/administrator/AdministratorActorSpec.scala:71]: Or maybe `contain only(actualComponents)` or something

**Pull request closed without merging**